native/cocos/core/scene-graph/Node.h

```diff
@@ -357,10 +357,10 @@ class Node : public BaseNode {
     /**
      * @en Inversely transform a point from world coordinate system to local coordinate system.
      * @zh 逆向变换一个空间点，一般用于将世界坐标转换到本地坐标系中。
-     * @param out The result point in local coordinate system will be stored in this vector
      * @param p A position in world coordinate system
+     * @return The result point in local coordinate system will be stored in this vector
      */
-    void inverseTransformPoint(Vec3 &out, const Vec3 &p);
+    Vec3 inverseTransformPoint(const Vec3 &p);
```

Comment:
Why change it. It is different from TS implementation.

Reply:
There are two main reasons:

1. `Vec3& out` is not user friendly for c++, c++ has stack for returning a temporary variable, but JS objects are allocated on heap, so JS needs a `out` parameter for returning value without generating GC objects.

2. JS binding could not bind a function with a `out` parameter, the original `inverseTransformPoint` auto binding code will not work properly.

Actually, we need to bind `inverseTransformPoint` manually. I will do it later.

Reply:
Ok. Will modify in this PR, right?

Reply:
No, I will create a new issue for that.

Reply:
https://github.com/cocos/cocos-engine/issues/11570